source/fundamentals/crud/compound-operations.txt

```diff
@@ -0,0 +1,313 @@
+===================
+Compound Operations
+===================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide you will learn how to perform **compound operations** with the
+MongoDB Java driver. 
+
+Compound operations consist of a read and write operation performed as one
+**atomic operation**. An atomic operation is an operation which either completes
+entirely, or does not complete at all. Atomic operations cannot partially complete.
+
+Atomic operations can help you avoid **race conditions** in your code. A
+race condition occurs when your code's behavior is dependent on the order of
+uncontrollable events.
+
+MongoDB supports the following compound operations:
+
+- Find and update one document
+- Find and replace one document
+- Find and delete one document
+
+If you need to perform more complex tasks atomically, such as reading and
+writing to more than one document, use **transactions**. Transactions are a
+feature of MongoDB and other databases that lets you define an arbitrary
+sequence of database commands as an atomic operation.  
+
+For more information on atomic operations and atomicity, see 
+:manual:`the MongoDB manual entry for atomicity and transactions </core/write-operations-atomicity/>`.
+
+For more information on transactions, 
+:manual:`see the MongoDB manual entry for transactions </core/transactions/>`.
+
+How to Use Compound Operations
+------------------------------
+
+This section shows how to use each compound operation with the MongoDB Java Driver.
+
+The following examples use a collection containing these two sample documents.
+
+.. code-block:: json
+
+    {"_id": 1, "food": "donut", "color": "green"}
+    {"_id": 2, "food": "pear", "color": "yellow"}
+
+`The full code for the following examples is available on Github here. <https://github.com/mongodb/docs-java/blob/master/source/includes/fundamentals/code-snippets/CompoundOperatorsIndividualExamples.java>`__
+
+.. note:: Before or After the Write?
+
+   By default, compound operations return your found document in the state
+   before your write operation. You can retrieve your found document in the
+   state after your write operation by using the options class corresponding to
+   your compound operation. You can see an example of this configuration in the 
+   :ref:`Find and Replace example below <java-find-and-replace-example>`.
+
```

Comment:
```suggestion
.. note:: Before or After the Write?

   By default, each compound operation returns your found document in the state
   before your write operation. You can retrieve your found document in the
   state after your write operation by using the options class corresponding to
   your compound operation. 

```

Reply:
Suggestion: Mentioning that you return the state of the document in each example by appending ``returnDocument(ReturnDocument.AFTER)`` to the option makes your note make more sense.  At first, I was confused why you put it so high up.

Doing this would also avoid having to explain it in each section that this is what you're doing.

Reply:
This comment was discussed synchronously. It was agreed that the example link is necessary as only one example discusses this option. We agreed that the earlier edit around "each compound returns"... improved the page.